### ISSUE TITLE
Object naming

### ISSUE THREAD

**Issue by lmoresi** [underworldcode/underworld2#149]
I'd like to have the option (as a user) to give each object a textual name which has meaning in the user space but which also would be in a standard location so we could document it / check for namespace clashes etc. 

Some obvious uses: naming variables so that it is possible to double check which data object has been given to a solver or is living on a particular swarm; naming visualisation objects so that the viewer does not have to use the internal, barely-readable automatic names; giving a better sense of interactivity when asking what something is than just getting back the object pointer.

L

**Comment by jmansour** [comment]
Yes, I think this is a good idea.   I'll add it to the post September milestone.

**Comment by jmansour** [comment]
Won't be done for UW2.  For UW3 we should be relatively close to this, as PETSc objects can be named, and similarly for Sympy.  I'll move this ticket there as it is currently relevant.
